text/ruby/0000-ruby-dependencies.md

```diff
@@ -0,0 +1,86 @@
+or in another fashion maintainers see fit. Regardless, the resulting dependency
+should work on both the Bionic and Jammy stack images, as well as be extendable
+to work with other stacks in the future.
+
+### Bundler (Bundler Buildpack)
+
+The `bundler` dependency will be kept as a Paketo-hosted dependency, since it
+is packaged as a Ruby `gem` file upstream, and needs to undergo some
+compilation / processing to be easily consumable in the buildpacks.
+
+New versions of bundler can be retrieved from
+https://rubygems.org/api/v1/versions/bundler.json, as well as the associated
+SHA256 and release date. The SHA256 can be used for dependency verification.
+Once the new versions are discovered, the dependency source can be retrieved
+from https://rubygems.org/downloads/bundler-VERSION.gem and used for
+compilation.
+
+### Curl (Passenger Buildpack)
+
+The `curl` dependency used in the Passenger buildpack will also be a
+Paketo-hosted dependency. Even though there are hosted Linux
+distribution-agnostic binaries available at https://curl.se/download.html, they
+are not provided by trusted sources that we should leverage in the project.
+Ubuntu, which is a trusted source, also [provides up-to-date curl
+binaries](https://packages.ubuntu.com/bionic/curl ), but they are provided as
+`.deb` files which are not easily consumed within the buildpack.
+
+Instead of using the upstream dependency, `curl` will be compiled from source.
+New versions will be discovered from
+[Github](https://github.com/curl/curl/releases), rather than a Curl CSV
+document, which is used [in the
+dep-server](https://github.com/paketo-buildpacks/dep-server/blob/f20264702d4010407c54d0c0d2a69186d9d324cf/pkg/dependency/curl.go#L83).
+The source used for compilation can be downloaded from
+https://curl.se/download/curl-VERSION.tar.gz. The dependency will need to be
+compiled in way that will be compatible with the Bionic stack, Jammy stack, and
+other future stacks.
+
+Each `curl` dependency has a PGP signature that can be retrieved from
+https://curl.se/download/curl-VERSION.tar.gz.asc. In legacy [dep-server
+code](https://github.com/paketo-buildpacks/dep-server/blob/f20264702d4010407c54d0c0d2a69186d9d324cf/pkg/dependency/curl.go#L157-L172)
+the signature is verified against https://daniel.haxx.se/mykey.asc, the public
+key of the author of `curl`. A similar verification step can be performed
+during compilation for dependency verification, however the public key should
+be downloaded and stored in the buildpack repository for usage for security
+purposes.  There are no SHA256 or other checksums published on the website or
+in Github releases, so PGP signature verification is the best option. A source
+SHA256 should still be manually calculated and added to the metadata for
+completeness.
+
+
+## Unresolved Questions and Bikeshedding
+- Is there a better solution for `curl`? Can we somehow leverage `apt` or
```

Comment:
While I think we could leverage `apt` to install curl and then copy the relevant files, I think it's probably about as complicated as installing from scratch ourselves, if not more complex.